src/modules/gfx_autoterminal.c

```diff
@@ -0,0 +1,523 @@
+    return 0;
+}
+
+static void parse_csi(char *str, int end){
+    int i = 1;// skip [
+    int j = 0;
+    int j_len = 0;
+    int code = 0;
+    switch(str[end]){
+    case 'm':
+        interpret_sgr(str, i);
+        break;
+    case 'A':
+        i = parse_sgr_value(str, i, &code, 1);
+        current_row -= i;
+        if(current_row < 0)
+            current_row = 0;
+        break;
+    case 'B':
+        i = parse_sgr_value(str, i, &code, 1);
+        current_row += i;
+        if(current_row > max_row)
+            current_row = max_row;
+        break;
+    case 'C':
+        i = parse_sgr_value(str, i, &code, 1);
+        current_column += i;
+        if(current_column > max_column)
+            current_column = max_column;
+        break;
+    case 'D':
+        i = parse_sgr_value(str, i, &code, 1);
+        current_column -= i;
+        if(current_column < 0)
+            current_column = 0;
+        break;
+    case 'H': // cursor position
+        i = parse_sgr_value(str, i, &code, 1);
+        current_row = code - 1;
+        // column handled by G
+    case 'G': // cursor horizontal absolute
+        parse_sgr_value(str, i, &code, 1);
+        current_column = code - 1;
+        break;
+    case 'K':// erase line
+        parse_sgr_value(str, i, &code, 0);
+        switch(code){
+        case 0:
+            j = current_column;
+            j_len = max_column;
+            break;
+        case 1:
+            j = 0;
+            j_len = current_column;
+            break;
+        case 2:
+            j = 0;
+            j_len = max_column;
+            break;
+        }
+        for(; j <= j_len; ++j){
+            buffer[j + (current_row * max_column)].c = ' ';
+            buffer[j + (current_row * max_column)].fg = fg;
+            buffer[j + (current_row * max_column)].bg = bg;
+        }
+        break;
+    case 'J'://erase in display
+        parse_sgr_value(str, i, &code, 0);
+        switch(code){
+        case 0:
+            j = current_column + (current_row * max_column);
+            j_len = max_column * max_row;
+            break;
+        case 1:
+            j = 0;
+            j_len = current_column + (current_row * max_column);
+            break;
+        default:
+            j = 0;
+            j_len = max_column * max_row;
+            break;
+        }
+        for(; j <= j_len; ++j){
+            buffer[j].c = ' ';
+            buffer[j].fg = fg;
+            buffer[j].bg = bg;
+        }
+        break;
+    default:
+        printf("Unhandled CSI type %c\n", str[end]);
+    }
+}
+
+// run in own thread
+// 6* is just a hack, since offset isn’t really working yet
+static void *launch(void *type_buffer) {
+    char *command = (char *)type_buffer;
+    int fd;
+    struct winsize win = {max_row, max_column, max_column * font_width, max_row * font_height};
+    if(forkpty(&fd, NULL, NULL, &win) == 0){
+        char* args[] = {"/bin/bash", "-c", command, NULL};
```

Comment:
Make that `/bin/sh`, please.
```suggestion
        char* args[] = {"/bin/sh", "-c", command, NULL};
```

Reply:
I think it’s better to read that via shebang from the script

Reply:
Yep.